### PR TITLE
Feature/si 2194 vehicle stream assignment

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

File: contracts/implementations/streamr/VehicleStream.sol [hunk 1]
```diff
@@ -62,7 +74,121 @@ contract VehicleStream is AccessControlInternal {
...
+
+        emit VehicleStreamAssociated(vehicleId, streamId);
+    }
+
+    /**
+     * @notice Allows a vehicle owner to dissociate the stream Id from their vehicle Id
+     * @dev Reverts if vehicle id does not exist or caller is not the owner
+     * @param vehicleId Vehicle node Id
+     */
+    function dissociateVehicleStream(uint256 vehicleId) external {
```
Comment (elffjs) [inline contracts/implementations/streamr/VehicleStream.sol:167]: It's a bit odd to me to have
`dissociateVehicleStream` -> `VehicleStreamDissociated`
but
`setVehicleStream` -> `VehicleStreamAssociated`

File: contracts/implementations/streamr/VehicleStream.sol [hunk 2]
```diff
@@ -62,7 +74,121 @@ contract VehicleStream is AccessControlInternal {
             )
         );
 
-        emit VehicleStreamAssociated(streamId, vehicleId);
+        string memory oldStreamId = vs.streams[vehicleId];
+        if (bytes(oldStreamId).length != 0) {
+            emit VehicleStreamDissociated(vehicleId, oldStreamId);
```
Comment (elffjs) [inline contracts/implementations/streamr/VehicleStream.sol:79]: @zer0stars Do we want to let people use the "default" stream if they already have another one set?

File: contracts/implementations/streamr/VehicleStream.sol [hunk 3]
```diff
@@ -62,7 +74,121 @@ contract VehicleStream is AccessControlInternal {
...
+            IStreamRegistry.PermissionType.Publish
+        );
+    }
+
+    /**
+     * @notice Allows a vehicle owner to associate an existing stream Id from their vehicle Id
+     * @dev This contract must have Publish and Grant Streamr permissions
+     * @dev Reverts if vehicle id does not exist, caller is not the owner,
+     * stream Id does not exist, DIMO Registry does not have grant permission or DIMO Streamr Node the publish permission
+     * @param vehicleId Vehicle node Id
```
Comment (elffjs) [inline contracts/implementations/streamr/VehicleStream.sol:98]: Missing `@param`?

File: contracts/implementations/streamr/VehicleStream.sol (hunk 2 above)
Comment (elffjs) [reply contracts/implementations/streamr/VehicleStream.sol:79]: @zer0stars Giving this a bump.

File: contracts/implementations/streamr/VehicleStream.sol (hunk 1 above)
Comment (zer0stars) [reply contracts/implementations/streamr/VehicleStream.sol:167]: VehicleStreamUnset?

File: contracts/implementations/streamr/VehicleStream.sol (hunk 2 above)
Comment (zer0stars) [reply contracts/implementations/streamr/VehicleStream.sol:79]: Not necessary.I would throw an error if they have one set. Then have them disassociate and reattach.

File: contracts/implementations/streamr/VehicleStream.sol (hunk 1 above)
Comment (LorranSutter) [reply contracts/implementations/streamr/VehicleStream.sol:168]: Let's do set and unset. Convers more cases.

**Review by elffjs — APPROVED** [review]

File: contracts/libraries/streamr/VehicleStreamStorage.sol [hunk 4]
```diff
@@ -0,0 +1,24 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+/**
+ * @title VechielStreamStorage
```
Comment (elffjs) [inline contracts/libraries/streamr/VehicleStreamStorage.sol:5]:
```suggestion
 * @title VehicleStreamStorage
```

File: utils/constants/Streamr.ts [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
+  ethers.toUtf8Bytes('TRUSTED_ROLE')
+);
+
+export const StreamrPermissionType = {
+  'Edit': 0,
+  'Delete': 1,
+  'Publish': 2,
+  'Subscribe': 3,
+  'Grant': 4
+}
```
Comment (elffjs) [inline utils/constants/Streamr.ts:17]: Newline ree.

**Review by elffjs — APPROVED** [review]

**Pull request merged**